mizar/networkpolicy/networkpolicy_util.py

```diff
@@ -327,7 +327,8 @@ def build_cidr_and_policies_map(self, access_rules, cidr_type):
             for found_cidr_tuple in found_cidr_map:
                 if indexed_policy_names != found_cidr_tuple[1]:
                     for foundPolicyName in found_cidr_tuple[1]:
-                        indexed_policy_names.add(foundPolicyName)
+                        if cidr in access_rules[cidr_and_policies_map_name] and foundPolicyName in access_rules[cidr_and_policies_map_name][cidr]:
```

Comment:
not sure why such (added) condition is needed. if some explanation is put here, that would help readers to understand.

Reply:
Added comments in code to explain why adding the line.